src/realm-settings/key-providers/aes-generated/AESGeneratedForm.tsx

```diff
@@ -0,0 +1,360 @@
+              type="text"
+              id="name"
+              name="name"
+              defaultValue={providerId}
+              validated={
+                form.errors.name
+                  ? ValidatedOptions.error
+                  : ValidatedOptions.default
+              }
+            />
+          </>
+        )}
+      </FormGroup>
+      <FormGroup
+        label={t("common:enabled")}
+        fieldId="kc-enabled"
+        labelIcon={
+          <HelpItem
+            helpText={t("realm-settings-help:enabled")}
+            forLabel={t("enabled")}
+            forID="kc-enabled"
+          />
+        }
+      >
+        <Controller
+          name="config.enabled"
+          control={form.control}
+          defaultValue={["true"]}
+          render={({ onChange, value }) => (
+            <Switch
+              id="kc-enabled"
+              label={t("common:on")}
+              labelOff={t("common:off")}
+              isChecked={value[0] === "true"}
+              data-testid={
+                value[0] === "true"
+                  ? "internationalization-enabled"
+                  : "internationalization-disabled"
+              }
+              onChange={(value) => {
+                onChange([value.toString()]);
+              }}
+            />
+          )}
+        />
+      </FormGroup>
+      <FormGroup
+        label={t("active")}
+        fieldId="kc-active"
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:active"
+            forLabel={t("active")}
+            forID="kc-active"
+          />
+        }
+      >
+        <Controller
+          name="config.active"
+          control={form.control}
+          defaultValue={["true"]}
+          render={({ onChange, value }) => {
+            return (
+              <Switch
+                id="kc-active"
+                label={t("common:on")}
+                labelOff={t("common:off")}
+                isChecked={value[0] === "true"}
+                data-testid={
+                  value[0] === "true"
+                    ? "internationalization-enabled"
+                    : "internationalization-disabled"
+                }
+                onChange={(value) => {
+                  onChange([value.toString()]);
+                }}
+              />
+            );
+          }}
+        />
+      </FormGroup>
+      <FormGroup
+        label={t("AESKeySize")}
+        fieldId="kc-aes-keysize"
+        labelIcon={
+          <HelpItem
+            helpText="realm-settings-help:AESKeySize"
+            forLabel={t("AESKeySize")}
+            forID="kc-aes-key-size"
+          />
+        }
+      >
+        <Controller
+          name="config.secretSize"
+          control={form.control}
+          defaultValue={["16"]}
+          render={({ onChange, value }) => (
+            <Select
+              toggleId="kc-aes-keysize"
+              onToggle={() => setIsKeySizeDropdownOpen(!isKeySizeDropdownOpen)}
```

Comment:
`onToggle` provides state so we do not have to manage it ourselves:

```suggestion
              onToggle={(isExpanded) => setIsKeySizeDropdownOpen(isExpanded)}
```